salt/netapi/rest_cherrypy/app.py

```diff
@@ -1899,6 +1899,8 @@ def POST(self, **kwargs):
 
             if token["eauth"] == "django" and "^model" in eauth:
                 perms = token["auth_list"]
+            elif token["eauth"] == "rest" and "auth_list" in token:
```

Comment:
The original issue would still remain unless the user has `keep_acl_in_token` set to True.

ping @dwoz any ideas here?

Reply:
@Ch3LL Yes but I don't immediately see how this would logically work, because if `keep_acl_in_token` is False, then the ACL must be fetched later. With config-defined permissions this is easy, as you can just read the file again. But with REST based back-ends they cannot be, as in a later context you don't have the password required to do another lookup.

Reply:
Thanks for the additional information, just waiting on @dwoz input to see if he has any ideas. If not we can progress this PR.

Reply:
I'm okay with this moving forward but I think we should force `keep_acl_in_token` to be true when using `rest` eauth backend and the behavior is clearly documented.

Reply:
Thanks! This has now been implemented.